source/core/parameters.cc

```diff
@@ -2375,13 +2383,179 @@ namespace Parameters
+                                    stoi(list_of_words_clean[0]);
+                                  std::string type = list_of_words_clean[1];
+                                  std::string arguments_str =
+                                    list_of_words_clean[2];
+                                  std::vector<std::string>
+                                    arguments_str_component =
+                                      Utilities::split_string_list(
+                                        arguments_str, ":");
+
+                                  unsigned int first_shape =
+                                    stoi(arguments_str_component[0]);
+                                  unsigned int second_shape =
+                                    stoi(arguments_str_component[1]);
+                                  if (type == "union")
+                                    {
+                                      operations[identifier] = std::make_tuple(
+                                        CompositeShape<
+                                          dim>::BooleanOperation::Union,
+                                        first_shape,
+                                        second_shape);
+                                    }
+                                  else if (type == "difference")
+                                    {
+                                      operations[identifier] = std::make_tuple(
+                                        CompositeShape<
+                                          dim>::BooleanOperation::Difference,
+                                        first_shape,
+                                        second_shape);
+                                    }
+                                  else if (type == "intersection")
+                                    {
+                                      operations[identifier] = std::make_tuple(
+                                        CompositeShape<
+                                          dim>::BooleanOperation::Intersection,
+                                        first_shape,
+                                        second_shape);
+                                    }
+                                }
+                            }
+                        }
+                      myfile.close();
+                      particles[i].shape =
+                        std::make_shared<CompositeShape<dim>>(
+                          components,
+                          operations,
+                          particles[i].position,
+                          particles[i].orientation);
+                    }
+                  else
+                    std::cout << "Unable to open file";
+                }
             }
           else
             {
               shape_arguments =
                 Utilities::string_to_double(shape_arguments_str_list);
+              particles[i].initialize_shape(shape_type, shape_arguments);
```

Comment:
The previous default value for composite (sphere) won't work here since it's not initialized. (I suggested to add the initialization in the default part).

Reply:
Good catch!